tools/mine-localisation-files.py

```diff
@@ -0,0 +1,208 @@
+    except:
+        print('Not a git repository')
+        return ''
+
+    if not re.fullmatch(r'git@github.com\:.*\/embedlite-components\.git', url):
+        print('Repository URL: {}'.format(url))
+
+    return root
+
+def convert_firefox_to_browser(filename):
+    content = ''
+    # Read the file contents
+    with open(filename, 'r') as file:
+        content = file.read()
+        content = re.sub(r'Firefox', 'Browser', content)
+    # Write out the new file contents
+    if content:
+        with open(filename, 'w') as file:
+            file.write(content)
+
+def get_path(filename, expected):
+    if os.path.isfile(expected + '/' + filename):
+        return expected + '/' + filename
+    possibilities = subprocess.check_output(['find', '.', '-iname', filename], encoding='ascii').split()
+    if len(possibilities) == 0:
+        # Nothing found
+        return ''
+
+    # The checks and their weightings
+    checks = []
+    # Check subfolders
+    checks.append((50, (lambda path : possibility.startswith(expected))))
+    # Check if path contains 'mobile'
+    checks.append((40, (lambda path : possibility.find('mobile') >= 0)))
+    # Check if path contains 'browser'
+    checks.append((30, (lambda path : possibility.find('browser') >= 0)))
+    # Check if path contains 'dom'
+    checks.append((20, (lambda path : possibility.find('dom') >= 0)))
+    # Check if path is non-empty
+    checks.append((10, (lambda path : len(path) > 0)))
+
+    # Find the best option based on the check weightings
+    best = ''
+    weight = 0
+    for possibility in possibilities:
+        for check in checks:
+            if check[0] > weight and check[1](possibility):
+                best = possibility
+                weight = check[0]
+
+    if not best:
+        # Just return the first one in the list
+        best = possibilities[0]
+
+    return best
+
+def convert_file(filename, expected, destination):
+    print('Copying and converting file: {}'.format(filename))
+
+    path = get_path(filename, expected)
+    print('Found version at: {}'.format(path))
+    if path:
+        final_path = destination + '/' + filename
+        print('Copying file to: {}'.format(final_path))
+        shutil. copyfile(path, final_path)
+        convert_firefox_to_browser(final_path)
+
+def mine_locale(locale, date):
+    repo_local = '/tmp/l10n-repo'
+    repo_remote = 'https://hg.mozilla.org/l10n-central/' + locale
+    retcode = 0
+    cloned = False
+
+    # Perform checks
+    git_root = get_git_root()
+    if not git_root:
+        print('You must execute this inside the embedlite-components git tree')
+        retcode = 128
+
+    if retcode == 0:
+        print('Git root: {}'.format(git_root))
+
+        cwd_orig = os.getcwd()
+
+        #os.chdir('/tmp')
+
+        # Clone the repository
+        print('Cloning repository {} to {}'.format(repo_remote, repo_local))
+        retcode = subprocess.call(['hg', 'clone', repo_remote, repo_local, '-U'])
```

Comment:
maybe add a comment to the script header which mentions that mercurial needs to be installed by the developer prior to running the script

Reply:
That's a good idea. I've added a comment, but also added in a simple check that will warn the user if either git or mercurial aren't installed.